diff --git a/render.py b/render.py
--- a/render.py
+++ b/render.py
@@ -101,18 +101,19 @@ def render_sets(dataset : ModelParams, iteration : int, pipeline : PipelinePara
         gaussians = GaussianModel(dataset)
         scene = Scene(dataset, gaussians, load_iteration=iteration, shuffle=False)
+        checkpoint_iteration = scene.loaded_iter
         if args.rgb_thermal and args.include_language:
-            checkpoint = os.path.join(args.model_path, 'thermal_language_chkpnt30000.pth')
+            checkpoint = os.path.join(args.model_path, f'thermal_language_chkpnt{checkpoint_iteration}.pth')
             (model_params, first_iter) = torch.load(checkpoint)
             gaussians.restore(model_params, args, mode='test')
         elif args.rgb_thermal:
-            checkpoint = os.path.join(args.model_path, 'thermal_chkpnt30000.pth')
+            checkpoint = os.path.join(args.model_path, f'thermal_chkpnt{checkpoint_iteration}.pth')
             (model_params, first_iter) = torch.load(checkpoint)
             gaussians.restore(model_params, args, mode='test')
         elif args.include_language:
-            checkpoint = os.path.join(args.model_path, 'language_chkpnt30000.pth')
+            checkpoint = os.path.join(args.model_path, f'language_chkpnt{checkpoint_iteration}.pth')
             (model_params, first_iter) = torch.load(checkpoint)
             gaussians.restore(model_params, args, mode='test')
         else:
-            checkpoint = os.path.join(args.model_path, 'chkpnt30000.pth')
+            checkpoint = os.path.join(args.model_path, f'chkpnt{checkpoint_iteration}.pth')
             (model_params, first_iter) = torch.load(checkpoint)
             gaussians.restore(model_params, args, mode='test')
 
