diff --git a/render.py b/render.py
--- a/render.py
+++ b/render.py
@@ -102,19 +102,19 @@ def render_sets(dataset : ModelParams, iteration : int, pipeline : PipelinePara
         scene = Scene(dataset, gaussians, load_iteration=iteration, shuffle=False)
         if args.rgb_thermal and args.include_language:
             checkpoint = os.path.join(args.model_path, 'thermal_language_chkpnt30000.pth')
-            (model_params, first_iter) = torch.load(checkpoint)
+            (model_params, first_iter) = torch.load(checkpoint, weights_only=False)
             gaussians.restore(model_params, args, mode='test')
         elif args.rgb_thermal:
             checkpoint = os.path.join(args.model_path, 'thermal_chkpnt30000.pth')
-            (model_params, first_iter) = torch.load(checkpoint)
+            (model_params, first_iter) = torch.load(checkpoint, weights_only=False)
             gaussians.restore(model_params, args, mode='test')
         elif args.include_language:
             checkpoint = os.path.join(args.model_path, 'language_chkpnt30000.pth')
-            (model_params, first_iter) = torch.load(checkpoint)
+            (model_params, first_iter) = torch.load(checkpoint, weights_only=False)
             gaussians.restore(model_params, args, mode='test')
         else:
             checkpoint = os.path.join(args.model_path, 'chkpnt30000.pth')
-            (model_params, first_iter) = torch.load(checkpoint)
+            (model_params, first_iter) = torch.load(checkpoint, weights_only=False)
             gaussians.restore(model_params, args, mode='test')
 
         bg_color = [1,1,1] if dataset.white_background else [0, 0, 0]
